SqlTrainer.UserService.Infrastructure/BusinessLogics/UserBusinessLogic.cs

```diff
@@ -12,4 +12,17 @@ public UserBusinessLogic(IUserRepository repository, IPasswordHasher passwordHas
         this.repository = repository;
         this.passwordHasher = passwordHasher;
     }
+
+    public async Task<IOperationResult<Guid>> RegisterAsync(User model)
+    {
+        var userResult = await repository.GetByLoginAsync(model.Login);
+
+        if (userResult.State == OperationResultState.Ok)
+            throw new Exception("User with the same login already exists");
+
+        var hashedPassword = passwordHasher.Hash(model.HashedPassword).Result;
```

Comment:
It should be in UserBusinessLogic and code above should be in RegisterService or smth like that